src/internal/git/fetch.go

```diff
@@ -48,3 +48,42 @@ func fetchTag(gitDirectory string, tag string) {
 		message.Fatal(err, "Not a valid tag or unable to fetch")
 	}
 }
+
+// FetchHash performs a `git fetch` of _only_ the provided commit hash.
+func FetchHash(gitDirectory string, hash string) {
+	message.Debugf("Fetch git hash %s from repo %s", hash, path.Base(gitDirectory))
+
+	repo, err := git.PlainOpen(gitDirectory)
+	if err != nil {
+		message.Fatal(err, "Unable to load the git repo")
+	}
+
+	remotes, err := repo.Remotes()
+	// There should never be no remotes, but it's easier to account for than
+	// let be a bug later
+	if err != nil || len(remotes) == 0 {
+		message.Fatal(err, "Failed to identify remotes.")
+	}
+
+	gitURL := remotes[0].Config().URLs[0]
+	message.Debugf("Attempting to find hash: %s for %s", hash, gitURL)
+
+	gitCred := FindAuthForHost(gitURL)
+
+	fetchOptions := &git.FetchOptions{
+		RemoteName: onlineRemoteName,
+		RefSpecs: []goConfig.RefSpec{
+			goConfig.RefSpec(hash + ":" + hash),
```

Comment:
It looks like this line (and an error check for `git.ErrTagExists` is the only difference between `FetchHash(...)` and `FetchTag(...)` 

Could we refactor this into a single `FetchRef(...)` function where we set the `RefSpecs` based on the output of the regexp you wrote in `pull.pull(..)`?